### PR TITLE
make metadata accessible for custom action_session_start

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/7420
- invalid slot configs now throw an exception which inherits from `RasaException`. This avoids noise in sentry reporting. 
- Fixed an error when using the endpoint `GET /conversations/<conversation_id:path>/story`
with a tracker which contained slots.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/processor.py [hunk 1]
```diff
@@ -184,12 +185,23 @@ async def _update_tracker_session(
                 f"Starting a new session for conversation ID '{tracker.sender_id}'."
             )
 
+            action_session_start = self._get_action(ACTION_SESSION_START_NAME)
+            if isinstance(
```
Comment (wochinge) [inline rasa/core/processor.py:189]: need to keep this as somebody might rely on it. No clue how to deprecate 😬

**Comment by akelad** [comment]
@wochinge do you need me to review anything in particular? I assume that Joe will do the actual code review

**Comment by wochinge** [comment]
@akelad Maybe give the docs a short glance?

File: rasa/core/processor.py [hunk 2]
```diff
@@ -184,12 +185,26 @@ async def _update_tracker_session(
                 f"Starting a new session for conversation ID '{tracker.sender_id}'."
             )
 
+            action_session_start = self._get_action(ACTION_SESSION_START_NAME)
+            # TODO: Remove in 3.0.0 and describe migration to `session_start_metadata`
+            # slot in migration guide.
+            if isinstance(
+                action_session_start, rasa.core.actions.action.ActionSessionStart
```
Comment (joejuzl) [inline rasa/core/processor.py:192]: This just determines that action_session_start has not been overriden by a custom action right?

File: rasa/shared/core/trackers.py [hunk 3]
```diff
@@ -907,7 +907,10 @@ def get_trackers_for_conversation_sessions(
 
     return [
         DialogueStateTracker.from_events(
-            tracker.sender_id, evts, tracker.slots, sender_source=tracker.sender_source
+            tracker.sender_id,
+            evts,
+            tracker.slots.values(),
```
Comment (joejuzl) [inline rasa/shared/core/trackers.py:912]: why this change?

**Review by joejuzl — APPROVED** [review]
Very nice and clean solution 👍

File: docs/docs/default-actions.mdx [hunk 4]
```diff
@@ -107,6 +107,32 @@ class ActionSessionStart(Action):
         return events
 ```
 
+If you want to access the metadata which was sent with the user message which triggered
+the session restart, you can access the special slot `session_started_metadata`:
```
Comment (akelad) [inline docs/docs/default-actions.mdx:111]:
```suggestion
the session start, you can access the special slot `session_started_metadata`:
```

?

**Comment by akelad** [comment]
docs look good to me 💯

File: rasa/shared/core/trackers.py (hunk 3 above)
Comment (wochinge) [reply rasa/shared/core/trackers.py:912]: It was a bug 😱 I'll add a changelog

File: rasa/core/processor.py (hunk 2 above)
Comment (wochinge) [reply rasa/core/processor.py:192]: yes because in that case it doesn't make sense to add metadata

File: docs/docs/default-actions.mdx (hunk 4 above)
Comment (wochinge) [reply docs/docs/default-actions.mdx]: yes, was wondering about it during the. self-review as well

File: rasa/shared/core/trackers.py (hunk 3 above)
Comment (wochinge) [reply rasa/shared/core/trackers.py:912]: changelog and test added
